PWGHF/TableProducer/trackToCollisionAssociator.cxx

```diff
@@ -0,0 +1,148 @@
+#include "Common/DataModel/TrackSelectionTables.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+using namespace o2::aod;
+
+struct HfTrackToCollisionAssociation {
+  Produces<HfTrackAssoc> association;
+  Produces<HfTrackCompColls> reverseIndices;
+
+  Configurable<float> nSigmaForTimeCompat{"nSigmaForTimeCompat", 3.f, "number of sigmas for time compatibility"};
+
+  using TracksWithSel = soa::Join<Tracks, TracksExtra, TrackSelection>;
+  Filter trackFilter = requireGlobalTrackWoDCAInFilter();
+
+  void processAssocWithTime(Collisions const& collisions,
+                            TracksWithSel const& tracksUnfiltered,
+                            soa::Filtered<TracksWithSel> const& tracks,
+                            BCs const& bcs)
+  {
+    std::vector<int>** collsPerTrack = new std::vector<int>*[tracksUnfiltered.size()];
+    memset(collsPerTrack, 0x0, sizeof(collsPerTrack) * tracksUnfiltered.size());
+
+    // loop over collisions to find time-compatible tracks
+    float nSigmaForTimeCompat2 = nSigmaForTimeCompat * nSigmaForTimeCompat;
+    auto trackBegin = tracks.begin();
+    const auto bOffsetMax = 241;
+    for (const auto& collision : collisions) {
+      float collTime = collision.collisionTime();
+      float collTimeRes2 = collision.collisionTimeRes() * collision.collisionTimeRes();
+      uint64_t collBC = collision.bc().globalBC();
+      bool iteratorMoved = false;
+      for (auto track = trackBegin; track != tracks.end(); ++track) {
+        if (!track.has_collision()) {
+          continue;
+        }
+        int64_t bcOffset = (int64_t)track.collision().bc().globalBC() - (int64_t)collBC;
+        float trackTime{0.};
+        float trackTimeRes2{0.};
+        if (track.isPVContributor()) {
+          trackTime = track.collision().collisionTime(); // if PV contributor, we assume the time to be the one of the collision
+          trackTimeRes2 = 25.f;                          // 1 BC
+        } else {
+          trackTime = track.trackTime();
+          trackTimeRes2 = track.trackTimeRes() * track.trackTimeRes();
+        }
+        float deltaTime = trackTime - collTime + bcOffset * 25.f;
+        float sigmaTimeRes2 = collTimeRes2 + trackTimeRes2;
+        LOGP(debug, "collision time={}, collision time res={}, track time={}, track time res={}, bc collision={}, bc track={}, delta time={}", collTime, collision.collisionTimeRes(), track.trackTime(), track.trackTimeRes(), collBC, track.collision().bc().globalBC(), deltaTime);
+        if (!iteratorMoved && bcOffset > -bOffsetMax) {
+          trackBegin.setCursor(track.filteredIndex());
+          iteratorMoved = true;
+          LOGP(debug, "Moving iterator begin {}", track.globalIndex());
+        } else if (bcOffset > bOffsetMax) {
+          LOGP(debug, "Stopping iterator {}", track.globalIndex());
+          break;
+        }
+        if (deltaTime * deltaTime < nSigmaForTimeCompat2 * sigmaTimeRes2) {
```

Comment:
As discussed in case of the 25, it should use 1 sigma here. So one needs a case. This case should also be used if the flag `TrackTimeResIsRange` is set 
https://github.com/AliceO2Group/AliceO2/blob/dev/Framework/Core/include/Framework/DataTypes.h